### ISSUE TITLE
tcareg won't finish after 24 hours

### ISSUE THREAD

**Issue by xinming-an** [cozygene/TCA#19]
the tcareg function cannot finish even after 24 hours of running. My sample size is less than 700. Below are the logs from the package. Any suggestions are greatly appreciated. Thank you! 

INFO [2023-10-04 09:17:36] Starting tca...
INFO [2023-10-04 09:17:37] Validating input...
INFO [2023-10-04 09:18:07] Fitting the TCA model...
INFO [2023-10-04 09:18:08] Fitting means and variances...
INFO [2023-10-04 09:18:08] Iteration 1 out of 10 internal iterations...

  |                                                        
  |                                                  |   0%
  |                                                        
  |+++++                                             |  10%
  |                                                        
  |++++++++++                                        |  20%
  |                                                        
  |+++++++++++++++                                   |  30%
  |                                                        
  |++++++++++++++++++++                              |  40%
  |                                                        
  |+++++++++++++++++++++++++                         |  50%
  |                                                        
  |++++++++++++++++++++++++++++++                    |  60%
  |                                                        
  |+++++++++++++++++++++++++++++++++++               |  70%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++          |  80%
  |                                                        
  |+++++++++++++++++++++++++++++++++++++++++++++     |  90%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%

  |                                                        
  |                                                  |   0%
  |                                                        
  |+++++                                             |  10%
  |                                                        
  |++++++++++                                        |  20%
  |                                                        
  |+++++++++++++++                                   |  30%
  |                                                        
  |++++++++++++++++++++                              |  40%
  |                                                        
  |+++++++++++++++++++++++++                         |  50%
  |                                                        
  |++++++++++++++++++++++++++++++                    |  60%
  |                                                        
  |+++++++++++++++++++++++++++++++++++               |  70%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++          |  80%
  |                                                        
  |+++++++++++++++++++++++++++++++++++++++++++++     |  90%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%
INFO [2023-10-04 09:40:28] Iteration 2 out of 10 internal iterations...

  |                                                        
  |                                                  |   0%
  |                                                        
  |+++++                                             |  10%
  |                                                        
  |++++++++++                                        |  20%
  |                                                        
  |+++++++++++++++                                   |  30%
  |                                                        
  |++++++++++++++++++++                              |  40%
  |                                                        
  |+++++++++++++++++++++++++                         |  50%
  |                                                        
  |++++++++++++++++++++++++++++++                    |  60%
  |                                                        
  |+++++++++++++++++++++++++++++++++++               |  70%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++          |  80%
  |                                                        
  |+++++++++++++++++++++++++++++++++++++++++++++     |  90%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%

  |                                                        
  |                                                  |   0%
  |                                                        
  |+++++                                             |  10%
  |                                                        
  |++++++++++                                        |  20%
  |                                                        
  |+++++++++++++++                                   |  30%
  |                                                        
  |++++++++++++++++++++                              |  40%
  |                                                        
  |+++++++++++++++++++++++++                         |  50%
  |                                                        
  |++++++++++++++++++++++++++++++                    |  60%
  |                                                        
  |+++++++++++++++++++++++++++++++++++               |  70%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++          |  80%
  |                                                        
  |+++++++++++++++++++++++++++++++++++++++++++++     |  90%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%
INFO [2023-10-04 10:14:06] Internal loop converged.
INFO [2023-10-04 10:14:06] Finished tca.
[1] "tca model complete"
INFO [2023-10-04 10:14:06] Starting tcareg...
INFO [2023-10-04 10:14:06] Validating input...
INFO [2023-10-04 10:14:13] Preparing data...
INFO [2023-10-04 10:14:29] Running TCA regression...
INFO [2023-10-04 10:14:29] Estimate tensor...

  |                                                        
  |                                                  |   0%
  |                                                        
  |+++++                                             |  10%
  |                                                        
  |++++++++++                                        |  20%
  |                                                        
  |+++++++++++++++                                   |  30%
  |                                                        
  |++++++++++++++++++++                              |  40%
  |                                                        
  |+++++++++++++++++++++++++                         |  50%
  |                                                        
  |++++++++++++++++++++++++++++++                    |  60%
  |                                                        
  |+++++++++++++++++++++++++++++++++++               |  70%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++          |  80%
  |                                                        
  |+++++++++++++++++++++++++++++++++++++++++++++     |  90%
  |                                                        
  |++++++++++++++++++++++++++++++++++++++++++++++++++| 100%
INFO [2023-10-04 14:12:22] Finished estimating tensor.

  |                                                        
  |                                                  |   0%

**Comment by E-R** [comment]
You never saw it goes above 0%?
Can you try running tcareg on a small set of CpGs (e.g., 100) and make sure it works?

**Comment by xinming-an** [comment]
Yes, the program never goes above 0% with the original data (around 700,000 CpGs). The function did work with 10,000 CpGs. Shall I split the data into small parts and apply the function to each?

**Comment by E-R** [comment]
What was the runtime of the tcareg command in the 10,000 cpgs run?
